File: Nearest Exit from Entrance in Maze/nearest-exit-from-entrance-in-maze.java
class Solution {
    public int nearestExit(char[][] maze, int[] entrance) {
        int n = maze.length;
        int m = maze[0].length;
        Queue<int[]> queue = new LinkedList<>();
        queue.offer(new int[]{entrance[0], entrance[1]});
        int steps = 0;

        while (!queue.isEmpty()) {
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                int[] current = queue.poll();
                int row = current[0], col = current[1];
                
                maze[row][col] = '+';

                if (!(row == entrance[0] && col == entrance[1]) &&
                    (row == 0 || row == n - 1 || col == 0 || col == m - 1)) {
                    return steps;
                }

                if (row - 1 >= 0 && maze[row - 1][col] != '+') {
                    queue.offer(new int[]{row - 1, col});
                    maze[row - 1][col] = '+';
                }
                if (row + 1 < n && maze[row + 1][col] != '+') {
                    queue.offer(new int[]{row + 1, col});
                    maze[row + 1][col] = '+';
                }
                if (col - 1 >= 0 && maze[row][col - 1] != '+') {
                    queue.offer(new int[]{row, col - 1});
                    maze[row][col - 1] = '+';
                }
                if (col + 1 < m && maze[row][col + 1] != '+') {
                    queue.offer(new int[]{row, col + 1});
                    maze[row][col + 1] = '+';
                }
            }
            steps++;
        }

        return -1;
    }
}

File: Nearest Exit from Entrance in Maze/nearest-exit-from-entrance-in-maze.py
class Solution(object):
    def nearestExit(self, maze, entrance):
        """
        :type maze: List[List[str]]
        :type entrance: List[int]
        :rtype: int
        """
        rows, cols = len(maze), len(maze[0])
        queue = deque([(entrance[0], entrance[1])])
        steps = 0
        
        while queue:
            size = len(queue)
            for _ in range(size):
                row, col = queue.popleft()
                
                maze[row][col] = '+'
                
                if (row != entrance[0] or col != entrance[1]) and \
                   (row == 0 or row == rows - 1 or col == 0 or col == cols - 1):
                    return steps
                
                for dr, dc in [(-1, 0), (1, 0), (0, -1), (0, 1)]:
                    nr, nc = row + dr, col + dc
                    if 0 <= nr < rows and 0 <= nc < cols and maze[nr][nc] == '.':
                        queue.append((nr, nc))
                        maze[nr][nc] = '+'
                        
            steps += 1
        
        return -1
